### ISSUE TITLE
Reduce background noise while recording.

### ISSUE THREAD

**Issue by AbhishekYadav03** [Canardoux/flutter_sound#956]
Normalize audio recording or reduce background noise while recording.

**Is your feature request related to a problem? Please describe.**
When I record sound from my flutter app it has a lot of background noise but when I recorded from the default recorder app the sound was very good.

**Describe the solution you'd like**
```
await _audioRecorder.startRecorder(
    codec: Codec.aacADTS,
    toFile: audioFile.path,
    audioSource: AudioSource.microphone,
    noiseReduction: true,
);
```

**Describe alternatives you've considered**
We can convert audio after the recording is completed using ffmpegKit

**Additional context**
This feature will increase sound recording quality.

**Comment by anukools** [comment]
@AbhishekYadav03 I could not find noiseReduction parameter ?

Whether you were able to reduce noise anyhow while recording ?

**Comment by AbhishekYadav03** [comment]
No, I'm not able to reduce the noise while recording.
The   `noiseReduction: true,` is a suggestion.

**Comment by JUANITOTELO** [comment]
You could try changing the codec for linear16pcm, with a sample rate of 41000, that works for me.

**Comment by ayesha-119** [comment]
@AbhishekYadav03 hello , i need your little help with one of your repo , sound recorder, can you please tell me how can i connect with you?

**Comment by alexgrusu** [comment]
any update on this?

**Comment by Larpoux** [comment]
Thanks to remind me this point. There are so many Open Issues that I am completely lost in this mess.
I restarted [Kanban Table](https://github.com/Canardoux/flutter_sound/projects/3) so that it will help me for the project Management.
Probably good if Flutter Sound users ping me to remind me that their issues are not yet solved.

For your point, you will see that your issue is in the `Todo in 10.0` column : actually I am overload with Flutter Sound 9.x maintenance, and I would like to release 10.0 in a not so much long time. So I prefer focus on 10.0 and not do enhancement in 9.x

**Comment by Mastermind-sap** [comment]
> I've noticed that default bit rate and sample size are really low.

@rezmeplxrf Hey could you tell me what values fixed your problem?? Like how much bitRate and sampleSize reduces the background noise??

**Comment by Larpoux** [comment]
Is this thread and #1172 on the same object ?

**Comment by Mastermind-sap** [comment]
@Larpoux Yes it is the same. Well increasing the bitRate and sampleSize to 96KHz helped reduce some of the noise. But the more I work on the audio it seems to lag and I get something like
```Skipped 116 frames!  The application may be doing too much work on its main thread.```.

Also there is huge difference in audio recorded indoors and outdoors, as far as I got to know while using.

**Comment by Larpoux** [comment]
[Taudio](https://taudio.canardoux.xyz/) 10.2.2 is released. I implemented two new boolean parameters in `startRecorder` : 
- enableNoiseSuppression
- enableEchoCancellation

These parameters are only implemented on Android. If someone desperately needs these parameters on other platforms, please fill an issue on Github

Note 1: I am not able to test the effect of these parameters. Just try and tell me.

Note 2:  It is very simple to upgrade to [Taudio](https://taudio.canardoux.xyz/) 10.2.2. Just 5 minutes. You can look to this [simple guide](https://taudio.canardoux.xyz/tau/fs2taudio.html)

**Comment by Mastermind-sap** [comment]
@Larpoux Thats great. I will give it a try and let you know if the effect is working.
It would be great if you could make it compatible with iOS also.

**Comment by unkindypie** [comment]
Hey @Larpoux when do you plan to release IOS?

**Comment by Larpoux** [comment]
I am going to look if can support those two new parameters on iOS tomorrow.

**Comment by truongdo** [comment]
@Larpoux  please make it for iOS 🥇 . I've been using flutter_sound for years and very happy about it, but now I run into a big issue with playing sound while recording :(

**Comment by olololoe110399** [comment]
@Larpoux please make it for iOS 🥇

**Comment by Larpoux** [comment]
Ok guys and girls,
I am going to see what I can do.

**Comment by olololoe110399** [comment]
> Ok guys and girls, I am going to see what I can do.

I waiting for you for finish repo: https://github.com/olololoe110399/ai_assistant
Currently, i stop record when playing from response audio.
And I try rtc sdk - agora with small code :v
